.eslintrc.js

```diff
@@ -2,98 +2,120 @@ module.exports = {
   },
 
   plugins: [],
-  extends: ['@xrplf/eslint-config/base', 'plugin:mocha/recommended'],
+  extends: ["@xrplf/eslint-config/base", "plugin:mocha/recommended"],
   rules: {
     // Certain rippled APIs require snake_case naming
-    '@typescript-eslint/naming-convention': [
-      'error',
+    "@typescript-eslint/naming-convention": [
+      "error",
       {
-        selector: 'interface',
-        format: ['PascalCase']
+        selector: "interface",
+        format: ["PascalCase"],
       },
       {
-        selector: 'interface',
-        format: ['snake_case']
-      }
+        selector: "interface",
+        format: ["snake_case"],
+      },
     ],
 
     // Ignore type imports when counting dependencies.
-    'import/max-dependencies': [
-      'error',
+    "import/max-dependencies": [
+      "error",
       {
         max: 5,
-        ignoreTypeImports: true
-      }
+        ignoreTypeImports: true,
+      },
     ],
     // Removes comments and blank lines from the max-line rules
-    'max-lines-per-function': [
-      'warn',
+    "max-lines-per-function": [
+      "warn",
       {
         max: 50,
         skipBlankLines: true,
-        skipComments: true
-      }
+        skipComments: true,
+      },
     ],
-    'max-lines': [
-      'warn',
+    "max-lines": [
+      "warn",
       {
         max: 250,
         skipBlankLines: true,
-        skipComments: true
-      }
-    ]
+        skipComments: true,
+      },
+    ],
   },
   overrides: [
     {
-      files: ['test/**/*.ts'],
+      files: ["test/**/*.ts"],
       rules: {
         // Removed the max for test files and test helper files, since tests usually need to import more things
-        'import/max-dependencies': 'off',
+        "import/max-dependencies": "off",
 
         // describe blocks count as a function in Mocha tests, and can be insanely long
-        'max-lines-per-function': 'off',
+        "max-lines-per-function": "off",
 
         // Tests can be very long turns off max-line count
-        'max-lines': 'off',
+        "max-lines": "off",
 
         // We have lots of statements in tests
-        'max-statements': 'off',
+        "max-statements": "off",
 
         // We have lots of magic numbers in tests
-        'no-magic-number': 'off'
-      }
+        "no-magic-number": "off",
+        "@typescript-eslint/no-magic-numbers": "off",
+
+        // We need to test things without type guards sometimes
+        "@typescript-eslint/no-unsafe-assignment": "off",
+
+        // We need to mess with internal things to generate certain testing situations
+        "@typescript-eslint/no-unsafe-member-access": "off",
+
+        // We need to be able to import xrpl-local
+        "node/no-extraneous-import": [
+          "error",
+          {
+            allowModules: ["xrpl-local"],
+          },
+        ],
+      },
+    },
+    {
+      files: ["test/models/*.ts"],
```

Comment:
When we're doing other linting tickets will we need to mess with the config like this? (Ex. when linting test/client)

Reply:
Perhaps - it depends on the specific errors you're getting. You should probably talk to @natenichols or me if you want to disable a rule for a whole folder.